app/features/transactions/transaction-list.tsx

```diff
@@ -207,15 +229,39 @@ export function TransactionList() {
     status,
   } = useTransactions();
 
+  const { data: notifications } = useNotifications({
+    type: 'PAYMENT_RECEIVED',
+  });
+  const { mutate: deleteAllNotifications } = useDeleteAllNotificationsByType({
+    type: 'PAYMENT_RECEIVED',
+  });
+
+  // If there are any notifications, delete them all on mount.
+  // We consider notifactions to be "read" as soon as the user opens the transaction list.
+  useEffect(() => {
+    deleteAllNotifications();
```

Comment:
is this how it should work? there is pagination on this page. what if currently ui is only showing first 10 transactions but there are 40 more with unread notifs? do we really want to make even those non visible ones as read?